src/BatchCommands.cpp

```diff
@@ -698,7 +698,6 @@ bool MacroCommands::ApplyMacro(
             undoManager.Undo(
                [&]( const UndoStackElem &elem ){
                   history.PopState( elem.state ); } );
-            undoManager.AbandonRedo();
```

Comment:
This shouldn't cause a problem: when reaching this stage, the macro has already pushed a new state, which means `AbandonRedo()` has already been called. At this stage, calling it again is trivial: there is nothing to undo already.